molgenis-core-ui/src/main/java/org/molgenis/core/ui/admin/permission/PermissionManagerController.java

```diff
@@ -605,8 +605,8 @@ else if (sid.equals(createAnonymousSid()))
 		}
 		else
 		{
-			String groupId = ((GrantedAuthoritySid) sid).getGrantedAuthority().substring("ROLE_".length());
-			permissions.setGroupId(groupId);
+			String roleId = ((GrantedAuthoritySid) sid).getGrantedAuthority().substring("ROLE_".length());
```

Comment:
Looks hacky, can we think of a solution that doesn't require String manipulation?

Reply:
Removed a lot of cruft from the Permissions model, it's become much simpler now and the stringfiddling is gone as well.